src/Features/Core/Portable/ExtractMethod/MethodExtractor.CodeGenerator.cs

```diff
@@ -316,18 +313,19 @@ protected ImmutableArray<TStatementSyntax> AppendReturnStatementIfNeeded(Immutab
             protected static HashSet<SyntaxAnnotation> CreateVariableDeclarationToRemoveMap(
                 IEnumerable<VariableInfo> variables, CancellationToken cancellationToken)
             {
-                var annotations = new List<(SyntaxToken, SyntaxAnnotation)>();
+                var annotations = new MultiDictionary<SyntaxToken, SyntaxAnnotation>();
```

Comment:
>MultiDictionary [](http://example.com/codeflow?start=38&length=15)

Need more coffee, what's the advantage of MultiDictionary in this case?

Reply:
Oh, I see, MethodExtractor.GetAnnotatedDocumentAndInsertionPointAsync is simplified by changing AddIdentifierTokenAnnotationPair to use a MultiDictionary

Reply:
It just more accurately reflects what the code was doing. It was building a list of K/V pairs, then later flattening to an IGrouping (or ILookup) of K to many Vs.